src/plugins/intel_cpu/src/nodes/kernels/jit_kernel_base.hpp

```diff
@@ -0,0 +1,216 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <string>
+#include <vector>
+#include <math.h>
+#include <dnnl_extension_utils.h>
+#include <cpu/x64/jit_generator.hpp>
+#include <cpu/x64/cpu_isa_traits.hpp>
+#include "registers_pool.hpp"
+#include "stack_allocator.hpp"
+
+namespace ov {
+namespace intel_cpu {
+
+using namespace dnnl::impl;
+namespace x64 = dnnl::impl::cpu::x64;
+
+class jit_kernel_base : public x64::jit_generator {
+public:
+    const size_t xmm_len = 16;
+    const size_t ymm_len = 32;
+    const size_t zmm_len = 64;
+
+    explicit jit_kernel_base(x64::cpu_isa_t max_cpu_isa)
+            : x64::jit_generator{nullptr, 256 * 1024, true, max_cpu_isa}
+            , max_cpu_isa_{max_cpu_isa} {}
+    virtual ~jit_kernel_base() {}
+
+    void generate() override {
+        this->preamble();
+        registers_pool_ = RegistersPool::create(max_cpu_isa_);
+        stack_allocator_ = std::make_shared<StackAllocator>(*this);
+        generate(registers_pool_, stack_allocator_);
+        registers_pool_.reset();
+        stack_allocator_.reset();
+        this->postamble();
+    }
+
+    virtual void generate(RegistersPool::Ptr registers_pool, StackAllocator::Ptr stack_allocator) {}
+
+    inline bool is_valid_isa(x64::cpu_isa_t isa) {
+        return x64::is_subset(isa, max_cpu_isa_) && x64::mayiuse(isa);
+    }
+
+    inline void uni_vgatherdps(const Xbyak::Xmm &xmm_val,
```

Comment:
How do you think of @dmitry-gorokhov 's suggestion in https://github.com/openvinotoolkit/oneDNN/pull/145 to use jit-emitter instead for gather and scatter?

Reply:
uni_vgatherdps() is not a part of this PR now